shared/middleware/helpers/bucket_helper.rb

```diff
@@ -98,8 +98,6 @@ def get(encrypted_channel_id, filename, if_modified_since = nil, version = nil)
       {status: 'NOT_MODIFIED'}
     rescue Aws::S3::Errors::NoSuchKey
       {status: 'NOT_FOUND'}
-    rescue Aws::S3::Errors::NoSuchVersion
-      {status: 'NOT_FOUND'}
```

Comment:
@epeach This was a mistake on my part!  Even though it's not documented, this is a real error and needs to be handled.  AWS differentiates between an invalid version (one that's not formatted correctly) and a missing version (right format, but not found).  I'll open a PR to put this back right away.

Reply:
Whoops, looks like we shipped this :cry:. I'm not sure of user bad here (404 vs 500, who cares?) but it might be noisy in honeybadger or elsewhere.  Heads-up @clareconstantine DOTD about this.

Reply:
Fix prepared here: https://github.com/code-dot-org/code-dot-org/pull/21284